app/models/employee.rb

```diff
@@ -62,17 +62,27 @@ def time_spent_on_tasks(employee_id, employee_tasks)
   end
 
   def compute_employees_velocity
-    @employees_list = company.teams.first.employees
+    if role == EMPLOYEE_ROLE
+      @employee_teams = teams
+    elsif role == ADMIN_ROLE
+      @employee_teams = company.teams
+    end
+
     @employee_tasks_data = {}
-    @employees_list.each do |employee|
-      @employee_tasks_data[employee.id] = {}
-      @employee_tasks_data[employee.id]['employee_id'] = employee.id
-      @employee_tasks_data[employee.id]['employee_name'] = employee.name
-      @employee_tasks = employee.company.tasks.get_employee_tasks(employee)
-      @employee_tasks_data[employee.id]['velocity'] = employee.compute_one_employee_velocity(employee.id, @employee_tasks)
-      @employee_tasks_data[employee.id]['total_tasks'] = @employee_tasks.count 
-      @employee_tasks_data[employee.id]['total_time'] = time_spent_on_tasks(employee.id, @employee_tasks)
-      @employee_tasks_data[employee.id]['total_complexity'] = compute_total_complexity_of_tasks(@employee_tasks)
+
+    @employee_teams.each do |team|
+      @employees_list = team.employees
+      @employee_tasks_data[team.id] = {}
+      @employees_list.each do |employee|
+        @employee_tasks_data[team.id][employee.id] = {}
+        @employee_tasks_data[team.id][employee.id]['employee_id'] = employee.id
```

Comment:
the way you are defining hash is wrong. 
It should be something like this.
`@employee_tasks_data[team.id][employee.id] = {
employee_id: employee.id,
employee_name: employee.name,
velocity: employee.compute_one_employee_velocity(employee.id, @employee_tasks)
}`

Reply:
changed!